### PR TITLE
Convert fulfill_a_promise to ES6 promise

### PR DESCRIPTION
**Pull request by TimothyGu**
@tdd, I've already started working on converting README etc. to ES6. See [diff](https://github.com/stevekane/promise-it-wont-hurt/compare/no-q).

This is the conversion of the first chapter.

### CODE REVIEW

**Comment by tdd** [comment]
Hey,

Hold off a second! I don't think converting the whole thing to ES6 only is a good thing. Promises are a valuable abstraction for anyone, including the ES5 world, and we should not require ES6 support for this workshopper to be available.

We should map the text to ES6, but make sure the workshopper runs on ES5 as well, through Jake's shim for instance, and explain that in the README and at the beginning of the first steps, repeatedly, so it sinks in.

**Comment by TimothyGu** [comment]
Yes I did offer a shim.

File: exercises/fulfill_a_promise/problem.md [hunk 1]
```diff
@@ -16,43 +16,65 @@ the syntax is.  We are about to dive into that in detail.
...
+the following in a Node.js REPL:
+
+```js
+typeof Promise !== "undefined"
+```
+
+If the REPL returns "true", then you are good to go! If not, you’ll have to use
+one of the many shims people have implemented. For this tutorial, we recommend
+`es6-promise` which aims to be strictly compliant to ES6 without any extra
+features. To use `es6-promise`, execute the following in a shell:
```
Comment (TimothyGu) [inline exercises/fulfill_a_promise/problem.md]: See?

**Comment by TimothyGu** [comment]
@tdd ping.

**Comment by tdd** [comment]
Hey @TimothyGu!  Sorry for the delay, I'm just over my head in a gazillion things these days: new office paperwork, firstborn due any day now, new employee, and loads more…

So, I went through your current PR state, updated the entire FR part for it, and did a few tiny tweaks (e.g. you had introduced a misspelling in "asynchronously" somewhere).

Note for later when you tweak FR text/solutions: in French, double punctuation (`!?:;`) always takes a space before it. Technically a half-width unbreakable space, which I usually just replace with an unbreakable space (Alt+Space on a Mac), but because workshop users don't necessarily know how to—or can—type this, I use regular ASCII space in anything they need to type. So I re-introduced the space before the bang in "RESOLU !" for the FR part.

That being said, this looks ready to merge back in `no-q`. Just ping me in the PR text for later challenges, and I'll do my best to follow suit ASAP with the translation.

Cheers!

**Comment by TimothyGu** [comment]
Congratulations on your baby! I totally understand how things can just seem to cramp together when they shouldn't be. To be honest, I was just asking for a code review and not a full French translation, as I understand that you have something much more important to take care of. Thank you for the mini French lesson as well.

**Comment by tdd** [comment]
Hey Tim,

Again, do go ahead with ES6 updates. I'll do my best to provide review and L10n along the way. It would be awesome to have a whole rework published before May 23 (Int'l NodeSchool Day)!

**Pull request merged**